tokio/src/task/local.rs

```diff
@@ -549,15 +549,23 @@ impl LocalSet {
                 .shared
                 .queue
                 .lock()
-                .pop_front()
+                .as_mut()
+                .and_then(|queue| queue.pop_front())
                 .or_else(|| self.context.tasks.borrow_mut().queue.pop_front())
         } else {
             self.context
                 .tasks
                 .borrow_mut()
                 .queue
                 .pop_front()
-                .or_else(|| self.context.shared.queue.lock().pop_front())
+                .or_else(|| {
+                    self.context
+                        .shared
+                        .queue
+                        .lock()
+                        .as_mut()
+                        .and_then(|queue| queue.pop_front())
```

Comment:
TIOLI: it seems like `queue.lock().as_mut().and_then(|queue| queue.pop_front())` _could_ be worth making a method on `Shared`, but :woman_shrugging: